docs/private-dns/connect-devices/mobile-and-desktop/linux.md

```diff
@@ -102,6 +102,21 @@ If you see a notification that you are not connected to AdGuard DNS, most likely
 
 :::
 
+## Use EDNS (Extended DNS)
+
+EDNS extends the DNS protocol, enabling larger UDP packets to carry additional data. In AdGuard DNS, it allows passing the DeviceID in plain DNS using an extra parameter.
+
+DeviceID, an eight-digit hexadecimal identifier (e.g., `1a2b3c4d`), helps link DNS requests to specific devices. For encrypted DNS, this ID is part of the domain (e.g., `1a2b3c4d.d.adguard-dns.com`). For unencrypted DNS, EDNS is required to localize this identifier.
+
+AdGuard DNS uses EDNS to retrieve the DeviceID. All you have to do is look for option number 65074 and, if such an option exists, read the DeviceID from there. For this, we need to use the command `dig` on the terminal:
+
+`dig @94.140.14.49 'www.example.com' A IN +ednsopt=65074:3031323334353637
+`
+
+Here, `65074` is the option ID, and `3031323334353637` is its value in hex format (DeviceID: 01234567)
+
+All done! The device ID will be displayed.
```

Comment:
I'd use “should”.  Who knows what kind of EDNS filtering could go on between the client and the server, heh.